osu.Game/Screens/Multi/Play/TimeshiftPlayer.cs

```diff
@@ -44,6 +53,16 @@ private void load()
 
             bool failed = false;
 
+            // Sanity checks to ensure that TimeshiftPlayer matches the settings for the current PlaylistItem
+            if (Beatmap.Value.BeatmapInfo.OnlineBeatmapID != playlistItem.Beatmap.OnlineBeatmapID)
+                throw new InvalidOperationException("Current Beatmap does not match PlaylistItem's Beatmap");
+
+            if (ruleset.Value.ID != playlistItem.Ruleset.ID)
+                throw new InvalidOperationException("Current Ruleset does not match PlaylistItem's Ruleset");
+
+            if (!CurrentMods.Value.Equals(playlistItem.RequiredMods))
```

Comment:
This will compare the `IEnumerable`s via reference. Likely what we want is something like:

```
foreach (var mod in playlistItem.RequiredMods)
    if (!CurrentMods.Value.Contains(mod))
        throw ...;
```